doc/subsystems/logging/logger.rst

```diff
@@ -159,6 +159,26 @@ appear in exactly one of them. Each other file should use
    #include <logging/log.h>
    LOG_MODULE_DECLARE(foo); /* In all files comprising the module but one */
 
+In order to use the logger in the header file,
```

Comment:
I see some technical comments that may change how this feature is implemented (and documented).  But here's a suggested rewrite of this opening paragraph if this implementation stays:

    Use the macro c:macro:`LOG_MODULE_DECLARE_IN_HEADER` if logging
    in a header file is needed.  Use of this macro must be preceded by a definition
    of ``LOG_IN_HEADER`` and ``LOG_LEVEL`` (as shown in the following example).
    Also note both defines must be undefined at the end of the header file: